pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,352 @@
+            brush = (0, 0, 255, 50)
+        self.setBrush(brush)
+
+        if hoverBrush is None:
+            hoverBrush = (0, 255, 255, 100)
+        self.setHoverBrush(hoverBrush)
+
+        self.currentPen = self.pen
+        self.currentBrush = self.brush
+
+        self._shape = None
+
+        self._pos = (0, 0)
+        if pos is None:
+            pos = (0, 0)
+        self.setPos(pos)
+
+        if isinstance(symbol, str):
+            try:
+                self._path = Symbols[symbol]
+            except KeyError:
+                raise KeyError("symbol name found in available Symbols")
+        elif isinstance(symbol, QtGui.QPainterPath):
+            self._path = symbol
+        else:
+            raise TypeError("Unknown type provided as symbol")
+        
+
+        self.scale = diameter
+        self.setPath(self._path)
+        self.setLabel(label, labelOpts)
+
+    def setPos(self, pos):
+        if isinstance(pos, tuple):
+            newPos = pos
+        elif isinstance(pos, list):
+            newPos = tuple(pos)
+        elif isinstance(pos, (QtCore.QPointF, QtCore.QPoint)):
+            newPos = (pos.x(), pos.y())
+        else:
+            raise TypeError
+        if self._pos != newPos:
+            self._pos = newPos
+            super().setPos(Point(self._pos))
+            self.sigPositionChanged.emit(self)
+
+    def setBrush(self, *args, **kwargs):
+        """Set the brush that fills the cursor. Allowable arguments are any that
+        are valid for :func:`~pyqtgraph.mkBrush`.
+        """
+        self.brush = fn.mkBrush(*args, **kwargs)
+        if not self.mouseHovering:
+            self.currentBrush = self.brush
+            self.update()
+
+    def setHoverBrush(self, *args, **kwargs):
+        """Set the brush that fills the cursor when hovering over it. Allowable
+        arguments are any that are valid for :func:`~pyqtgraph.mkBrush`.
+        """
+        self.hoverBrush = fn.mkBrush(*args, **kwargs)
+        if self.mouseHovering:
+            self.currentBrush = self.hoverBrush
+            self.update()
+
+    def setPen(self, *args, **kwargs):
+        """Set the pen for drawing the cursor. Allowable arguments are any that
+        are valid for :func:`~pyqtgraph.mkPen`."""
+        self.pen = fn.mkPen(*args, **kwargs)
+        if not self.mouseHovering:
+            self.currentPen = self.pen
+            self.update()
+
+    def setHoverPen(self, *args, **kwargs):
+        """Set the pen for drawing the cursor when hovering over it. Allowable
+        arguments are any that are valid for
+        :func:`~pyqtgraph.mkPen`."""
+        self.hoverPen = fn.mkPen(*args, **kwargs)
+        if self.mouseHovering:
+            self.currentPen = self.hoverPen
+            self.update()
+
+    def setLabel(self, text=None, labelOpts=None):
+        if text is None and self.label is not None:
+            # remove the label if it's already added
+            self.label.scene().removeItem(self.label)
```

Comment:
If I call `setLabel(None)` , this fails for me with
`AttributeError: 'NoneType' object has no attribute 'removeItem'`

If I call `setLabel('some text')` nothing happens, the default text does not change. No error message.

Reply:
Thanks for testing, was able to replicate each issue; fixed with e3d9c078ce871d25a40567353e44dc920a9e93d4